store/store.go

```diff
@@ -1445,8 +1459,12 @@ var applyDelta = func(name string, deltaPath string, deltaInfo *snap.DeltaInfo,
 
 	partialTargetPath := targetPath + ".partial"
 
+	xdelta3Bin, err := getXdelta3Binary()
+	if err != nil {
+		return err
+	}
 	xdelta3Args := []string{"-d", "-s", snapPath, deltaPath, partialTargetPath}
-	cmd := exec.Command("xdelta3", xdelta3Args...)
+	cmd := exec.Command(xdelta3Bin, xdelta3Args...)
```

Comment:
Can you please add a new function somewhere

`CommandFromCore(...)` that returns an exec.Command (trivially for now).

This function will need some love to make sure we can execute core executables on systems that differ in libc/shared libraries. I will implement this correctly later.